master/internal/api/error.go

```diff
@@ -62,3 +64,21 @@ func AsErrNotFound(msg string, args ...interface{}) error {
 		args...,
 	)
 }
+
+// APIErr2GRPC converts internal api error categories into grpc status.Errors.
+func APIErr2GRPC(err error) error {
+	switch {
+	case errors.Is(err, ErrBadRequest):
+		return status.Errorf(
+			codes.InvalidArgument,
+			err.Error(),
+		)
+	case errors.Is(err, ErrNotFound):
+		return status.Errorf(
+			codes.NotFound,
+			err.Error(),
+		)
+	default:
+		return nil
```

Comment:
I think default should just return the error, not discard it (so then the calls below are just `if err != nil { return api.APIErr2GRPC(err) }`. My concern is just this way is misuse-able and that is not.

Reply:
yeah, i see.. i just think this could get misused pretty easily and we could drop the error. in that case i guess it'd prefer 
`return api.APIErr2GRPC(errors.Wrap(err, "failed to prepare launch params"))`
I think the grpc error formed from the `error.Error()` of the passed error still needs to context "failed to prepare launch params" or else the error may just be like "string too long" or something (totally made up example)

Reply:
okay this works